File: backend/spotify-server/Models/Track/Track.cs
namespace spotify_server.Models.Track;

public class Track : IDefaultItem
{
    public string type {get;set;}
    public long id {get;set;}
    public long[] owner {get;set;}
    public string title {get;set;}
    public string image {get;set;}
    public string audio {get;set;}
    public DateTime date {get;set;}
}

File: frontend/src/models/Items/Track.ts
import DefaultItem from "../DefaultItem";
import ID from "../ID";
import {AudioURL, ImageURL} from "../url";
import ItemType from "../ItemType";
import { ArtistID, ArtistView } from "./Artist";

interface Track extends DefaultItem {
    type: ItemType.TRACK;
    id: TrackID;
    title: string;
    owner: ArtistView[];
    audio: AudioURL;
    image?: ImageURL;
    subtitles?: string;
}

export type TrackID = ID;

export default Track;